antropy/fractal.py

```diff
@@ -183,14 +184,37 @@ def katz_fd(x, axis=-1):
     1.0000
     """
     x = np.asarray(x)
-    dists = np.abs(np.diff(x, axis=axis))
-    ll = dists.sum(axis=axis)
-    ln = np.log10(ll / dists.mean(axis=axis))
-    aux_d = x - np.take(x, indices=[0], axis=axis)
-    d = np.max(np.abs(aux_d), axis=axis)
-    kfd = np.squeeze(ln / (ln + np.log10(d / ll)))
+
+    # euclidian distance calculation
+    euclidean_distance = np.sqrt(1 + np.square(np.diff(x, axis=axis)))
+
+    # total and average path lengths
+    total_path_length = euclidean_distance.sum(axis=axis)
+    average_path_length = euclidean_distance.mean(axis=axis)
+
+    # max distance from first to all
+    horizontal_diffs = np.arange(1, x.shape[axis])
+    vertical_diffs = np.take(x, indices=np.arange(1, x.shape[axis]), axis=axis) - np.take(
+        x, indices=[0], axis=axis
+    )
+
+    if axis == 1:  # reshape if needed
+        horizontal_diffs = horizontal_diffs.reshape(1, -1)
+    elif axis == 0:
+        horizontal_diffs = horizontal_diffs.reshape(-1, 1)
+
+    # Euclidean distance and max distance
+    distances = np.sqrt(np.square(horizontal_diffs) + np.square(vertical_diffs))
+    max_distance = np.max(distances, axis=axis)
+
+    # Katz Fractal Dimension Calculation
+    full_distance = np.log10(total_path_length / average_path_length)
+    kfd = np.squeeze(full_distance / (full_distance + np.log10(max_distance / total_path_length)))
+
+    # ensure scalar output
```

Comment:
Hi @PiethonProgram,

I think that this implementation can be simplified, for example by following the proposed new implementation in: https://github.com/raphaelvallat/antropy/issues/34, or by leveraging the Neurokit2 implementation (which as of present gives the same output as Antropy): https://github.com/neuropsychology/NeuroKit/blob/45c9ad90d863ebf4e9d043b975a10d9f8fdeb06b/neurokit2/complexity/fractal_katz.py#L6

Reply:
Sounds good, I will make the adjustments.

Reply:
Hello, I have taken a look at the implementations you mentioned. Are you sure it can be simplified? Previous implementations that you mentioned are shorter since they are all single-channel. 
If you want to only offer single-channel feature extraction then I can make the changes, but otherwise, unless you want to try and decrease time using Numba, I don't think there is much I can "simplify."

Reply:
Good point about the support for ND arrays. I have not yet found the time to do a deep dive into this, but can we just take the existing implementation of Antropy (see below) and replace the distance calculation by the Euclidean distance?

```python
dists = np.abs(np.diff(x, axis=axis))
ll = dists.sum(axis=axis)
ln = np.log10(ll / dists.mean(axis=axis))
aux_d = x - np.take(x, indices=[0], axis=axis)
d = np.max(np.abs(aux_d), axis=axis)
kfd = np.squeeze(ln / (ln + np.log10(d / ll)))
```

or is there more to your implementation that I'm missing?

Thank you

Reply:
I think the essence of the code is the same, but some additional "bits" are needed when using Euclidean distance in n-dimensions in order to check for distances from one to the other. 
If we were only speaking in 1-dimension, then yes, we can simply just replace the distance calculation line.